spec/policies/message_policy_spec.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe MessagePolicy do
+  subject { described_class.new(user, message) }
+
+  let(:user) { nil }
+  let(:recipient) { create(:user) }
+  let(:sender) { create(:user) }
+  let(:message) { create(:message, sender:, recipient:) }
+  let(:role) { 'member' }
+
+  context 'with a user' do
+    let(:user) { create(:user) }
+
+    it { is_expected.to permit_only_actions(%i[index new]) }
+
+    context 'when user is admin' do
+      let(:user) { create(:admin) }
+
+      it { is_expected.to permit_all_actions }
+    end
+
+    context 'when user is recipient' do
+      let(:user) { recipient }
+
+      it { is_expected.to permit_only_actions(%i[index new destroy]) }
```

Comment:
Why is this user not to reply? Or, phrasing this question differently: If I read the specs correctly, no one except an admin is allowed to reply. Why? Maybe we are just missing some specs here...

Reply:
Yes we are missing some specs here. The thing is that the reply permission for a message does not mean you are allowed to reply **to** this message but that you are allowed to reply **with** this message. Since you cannot send replies with yourself as recipient the user is not given the reply permission here. I will write another test covering the reply action for common users.

Reply:
Ah, I see. Thanks for explaining and adding another spec there!